cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/FlushToStorageCommand.java

```diff
@@ -46,35 +53,83 @@ public FlushToStorageCommand(CommandArgs args) {
 
     @Override
     public void execute() throws Exception {
-        ensureArgCount(2);
-
+        validateArguments();
         final String containerId = getArg(0);
-        final String segmentStoreHost = getArg(1);
+        int startContainerId;
+        int endContainerId;
         @Cleanup
         CuratorFramework zkClient = createZKClient();
         @Cleanup
         AdminSegmentHelper adminSegmentHelper = instantiateAdminSegmentHelper(zkClient);
         if (containerId.equalsIgnoreCase(ALL_CONTAINERS)) {
-            int containerCount = getServiceConfig().getContainerCount();
-            for (int id = 0; id < containerCount; id++) {
-                flushContainerToStorage(adminSegmentHelper, id, segmentStoreHost);
-            }
+            startContainerId = 0;
+            endContainerId = getServiceConfig().getContainerCount() - 1;
         } else {
-            flushContainerToStorage(adminSegmentHelper, parseInt(containerId), segmentStoreHost);
+            startContainerId = parseInt(containerId);
+            endContainerId = getArgCount() == 2 ? parseInt(getArg(1)) : startContainerId;
+        }
+
+        for (int id = startContainerId; id <= endContainerId; id++) {
+            flushContainerToStorage(adminSegmentHelper, id);
         }
     }
 
-    private void flushContainerToStorage(AdminSegmentHelper adminSegmentHelper, int containerId, String segmentStoreHost) throws Exception {
+    private void flushContainerToStorage(AdminSegmentHelper adminSegmentHelper, int containerId) throws Exception {
         CompletableFuture<WireCommands.StorageFlushed> reply = adminSegmentHelper.flushToStorage(containerId,
-                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
+                new PravegaNodeUri(this.getHosts().get(containerId), getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
         reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
         output("Flushed the Segment Container with containerId %d to Storage.", containerId);
     }
 
     public static CommandDescriptor descriptor() {
         return new CommandDescriptor(COMPONENT, "flush-to-storage", "Persist the given Segment Container into Storage.",
-                new ArgDescriptor("container-id", "The container Id of the Segment Container that needs to be persisted, " +
-                        "if given as \"all\" all the containers will be persisted."),
-                new ArgDescriptor("segmentstore-endpoint", "Address of the Segment Store we want to send this request."));
+                new ArgDescriptor("start-container-id", "The start container Id of the Segment Container that needs to be persisted, " +
+                        "if given as \"all\" all the containers will be persisted."), new ArgDescriptor("end-container-id", "The end container Id of the Segment Container that needs to be persisted, " +
```

Comment:
looking at this description its not clear that it can accept a single container. i.e flush-to-storage <a_given_container_id>

Reply:
> looking at this description its not clear that it can accept a single container. i.e flush-to-storage <a_given_container_id>

I have updated it.